src/buffer/out/sgrStack.hpp

```diff
@@ -0,0 +1,238 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- sgrStack.hpp
+
+Abstract:
+- Encapsulates logic for the XTPUSHSGR / XTPOPSGR VT control sequences, which save and
+  restore text attributes on a stack.
+
+--*/
+
+#pragma once
+
+#include "TextAttribute.hpp"
+#include "..\..\terminal\adapter\DispatchTypes.hpp"
+
+namespace Microsoft::Console::VirtualTerminal
```

Comment:
Given it's a VT-related support tool, I feel like it should belong to the /terminal/adapter library or, failing that, the /types/ library.

I'd rather it not live in /buffer/out/.

Reply:
Considering it needs to be used by both the host's adapter and the terminal's adapter, I'd probably recommend `/types/`, since that lib is used by both of them.

Reply:
Thanks--put it there.